service/matching/db_task_writer.go

```diff
@@ -107,10 +107,14 @@ func (f *dbTaskWriterImpl) appendTask(
 
 func (f *dbTaskWriterImpl) flushTasks(
 	ctx context.Context,
-) {
+) bool {
+	if len(f.taskBuffer) == 0 {
+		return false
+	}
 	for len(f.taskBuffer) > 0 {
 		f.flushTasksOnce(ctx)
 	}
+	return true
```

Comment:
instead of returning a boolean, try https://github.com/temporalio/temporal/pull/3204/files#diff-36a9587062acd592d13301c653abe1ab5edb2d7d260466e5be78b6440c61b3cfR90?